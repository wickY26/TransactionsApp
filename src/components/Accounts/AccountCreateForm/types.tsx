import { Account } from '../../../containers/Accounts/types';

export interface AccountCreateFormProps {
  onCreateAccount: (data: Account) => void;
}
