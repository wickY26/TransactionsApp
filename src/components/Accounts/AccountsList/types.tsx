import { Account } from '../../../containers/Accounts/types';

export interface AccountsListProps {
  accounts: Account[];
};
