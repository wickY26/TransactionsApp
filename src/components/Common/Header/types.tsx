export interface HeaderProps {
  title: string;
  link?: string;
  backIcon?: boolean;
}
