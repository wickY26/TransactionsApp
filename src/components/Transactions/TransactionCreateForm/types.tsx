import { TransactionBody } from '../../../containers/Transactions/types';

export interface TransactionCreateFormProps {
  onCreateTransaction: (data: TransactionBody) => void;
}
