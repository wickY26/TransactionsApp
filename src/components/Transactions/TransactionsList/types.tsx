import { Transaction } from '../../../containers/Transactions/types';

export interface TransactionsListProps {
  transactions: Transaction[];
};
