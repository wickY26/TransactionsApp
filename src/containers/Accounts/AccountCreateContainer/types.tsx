import { ReactElement } from 'react';

import { Account } from '../types';

export interface RenderProps {
  onCreateAccount: (data: Account) => void;
}

export interface AccountCreateContainerProps {
  children: (props: RenderProps) => ReactElement;
};
