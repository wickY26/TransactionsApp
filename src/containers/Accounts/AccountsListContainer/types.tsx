import { ReactElement } from 'react';

import { Account } from '../types';

export interface RenderProps {
  accounts: Account[];
}

export interface AccountsListContainerProps {
  children: (props: RenderProps) => ReactElement;
};
