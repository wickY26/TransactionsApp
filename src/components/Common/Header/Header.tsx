import React, { FC } from 'react';
import { Link } from 'react-router-dom';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import AddIcon from '@material-ui/icons/Add';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';

import { HeaderProps } from './types';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';

const Header: FC<HeaderProps> = ({ title, link, backIcon }) => (
  <>
    <Box display="flex" m={2} mr={0} alignItems="center">
      <Box display="flex" flexGrow={1}>
        <Typography variant="h4">
          {title}
        </Typography>
      </Box>
      {
        link && <Link to={link}>
          <IconButton>
            {
              backIcon ?
              <ArrowBackIcon color="primary" /> :
              <AddIcon color="primary" /> 
            }
          </IconButton>
        </Link>
      }
    </Box>
    <Divider />
  </>
);

export default Header;
