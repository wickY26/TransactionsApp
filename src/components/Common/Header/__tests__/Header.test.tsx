import React from 'react';
import { render } from '@testing-library/react';

import Header from '..';
import { HeaderProps } from '../types';

jest.mock('react-router-dom', () => ({
  Link: 'div',
}));

const renderComponent = (props?: Partial<HeaderProps>) => render(
  <Header title="Title"  {...props} />
);

describe('Header', () => {

  it('should render correctly', () => {
    const component = renderComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render link', () => {
    const component = renderComponent({ link: '/link'});
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render back icon', () => {
    const component = renderComponent({ link: '/link', backIcon: true });
    expect(component.asFragment()).toMatchSnapshot();
  });

});
