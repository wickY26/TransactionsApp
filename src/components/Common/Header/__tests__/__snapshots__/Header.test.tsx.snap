// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render back icon 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-5"
  >
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Title
      </h4>
    </div>
    <div
      to="/link"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <hr
    class="MuiDivider-root"
  />
</DocumentFragment>
`;

exports[`Header should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Title
      </h4>
    </div>
  </div>
  <hr
    class="MuiDivider-root"
  />
</DocumentFragment>
`;

exports[`Header should render link 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Title
      </h4>
    </div>
    <div
      to="/link"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <hr
    class="MuiDivider-root"
  />
</DocumentFragment>
`;
